lib/processors/index.js

```diff
@@ -0,0 +1,69 @@
+
+/**
+ * Dependencies
+ */
+
+var debug = require('debug')('processor');
+var parseContentType = require('content-type').parse;
+var request = require('superagent');
+
+/**
+ * User-Agent string for requests.
+ *
+ * @type {String}
+ */
+var ua = 'Mozilla/5.0 (Windows NT 6.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0.2228.0 Safari/537.36'; // jshint ignore:line
+
+/**
+ * Supported content-type processors.
+ *
+ * @type {Object}
+ */
+var processors = {
+  'text/html': require('./text-html'),
+  'text/calendar': require('./text-calendar')
+};
+
+module.exports = function(url) {
+  debug('processing', url);
+  return get(url)
+    .then(res => {
+      var type = getType(res);
+      var processor = processors[type];
+
+      if (!type) throw new Error('unknown type: ' + type);
```

Comment:
This will kill the server, should we log the error and return a null value, or something indicating that url could not be processed?